src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using NuGet.SolutionRestoreManager;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.NuGet
+{
+    internal static class ProjectRestoreInfoBuilder
+    {
+        private const string DefiningProjectDirectoryProperty = "DefiningProjectDirectory";
+        private const string ProjectFileFullPathProperty = "ProjectFileFullPath";
+
+        internal static IVsProjectRestoreInfo Build(IEnumerable<IProjectValueVersions> updates)
+        {
+            Requires.NotNull(updates, nameof(updates));
+
+            return Build(updates.Cast<IProjectVersionedValue<IProjectSubscriptionUpdate>>());
+        }
+
+        internal static IVsProjectRestoreInfo Build(IEnumerable<IProjectVersionedValue<IProjectSubscriptionUpdate>> updates)
+        {
+            Requires.NotNull(updates, nameof(updates));
+
+            // if none of the underlying subscriptions have any changes
+            if (!updates.Any(u => u.Value.ProjectChanges.Any(c => c.Value.Difference.AnyChanges)))
+            {
+                return null;
+            }
+
+            string baseIntermediatePath = null;
+            var targetFrameworks = new TargetFrameworks();
+            
+            foreach (IProjectVersionedValue<IProjectSubscriptionUpdate> update in updates)
+            {
+                var configurationChanges = update.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
+                baseIntermediatePath = baseIntermediatePath ?? 
+                    configurationChanges.After.Properties[ConfigurationGeneral.BaseIntermediateOutputPathProperty];
+                string targetFrameworkMoniker = 
+                    configurationChanges.After.Properties[ConfigurationGeneral.TargetFrameworkMonikerProperty];
+
+                if (targetFrameworks.Item(targetFrameworkMoniker) == null)
```

Comment:
If I understand this correctly, let's say I have this:

<TargetFrameworks>net45;net45_bar</TargetFrameworks>

and both result in a TFM of .NET Framework 4.5, you will ignore the second set of package and project references, even though build, intellisense and everything else will work. I don't think that's the correct behavior. We shouldn't be throwing away data @nguerrera